utilities/requestParser.js

```diff
@@ -1,34 +1,70 @@
 const _ = require('lodash');
+const moment = require('moment');
+const momentRange = require('moment-range');
 const queryString = require('query-string');
 
 const isParamValid = (inputParam, paramValidators) =>
-  _.every(_.values(paramValidators), validate => validate(inputParam));
+  _.every(_.values(paramValidators), (validate) => validate(inputParam));
 
 const validateInputParams = (inputParams, inputFormat) =>
   _.reduce(
     inputParams,
     (isValid, userInputProperty, propertyKey) =>
       isParamValid(userInputProperty, inputFormat[propertyKey]) && isValid,
-    true,
+    true
   );
 
+const getDatesBetween = (minDate, maxDate) => {
+  const min = moment(minDate, 'YYYY/MM/DD');
+
+  const max = moment(maxDate, 'YYYY/MM/DD');
+
+  if (
+    !min.isValid() &&
+    !max.isValid() &&
+    !min.isSameOrAfter(moment(), 'date') &&
+    !max.isSameOrAfter(moment(), 'date')
+  ) {
+    return [];
+  }
+  return _.map(
+    min.isSameOrBefore(max)
+      ? [ ...momentRange.extendMoment(moment).range(min, max).reverseBy('days') ]
+      : [ ...momentRange.extendMoment(moment).range(max, min).reverseBy('days') ],
+    (m) => m.format('YYYY/MM/DD')
+  );
+};
+
+const parseDatesToArray = (dates) => {
+  const [ minDate, maxDate ] = _.split(dates, '-');
+  if (!maxDate) {
+    const min = moment(minDate, 'YYYY/MM/DD');
+    return min.isValid && min.isSameOrAfter(moment(), 'date') ? [ min.format('YYYY/MM/DD') ] : [];
```

Comment:
This is the next place that `min.isValid && min.isSameOrAfter(moment(), 'date')` is checked. (23-26 lines too)

I assume that `min.isValid && min.isSameOrAfter(moment(), 'date')` is the key validity check, so maybe we can extract that to some external method or even `dateService` ?